src/bindings/swig/ruby/kdb.i

```diff
@@ -0,0 +1,691 @@
+%ignore kdb::Key::getStringSize;
+/* kdb::Key::getBinarySize could be useful */
+
+
+/* predicate methods rename to "is_xxx?" and return Rubys boolean */
+%predicate kdb::Key::isValid;
+%predicate kdb::Key::isSystem;
+%predicate kdb::Key::isUser;
+%predicate kdb::Key::isString;
+%predicate kdb::Key::isBinary;
+%predicate kdb::Key::isInactive;
+%predicate kdb::Key::isBelow;
+%predicate kdb::Key::isBelowOrSame;
+%predicate kdb::Key::isDirectBelow;
+%predicate kdb::Key::hasMeta;
+%predicate kdb::Key::isNull; // TODO: do we need something special here??? 
+%predicate kdb::Key::needSync;
+
+/* 
+ * be more Ruby native: 
+ * for all methods, which return a Key, for which Key.is_null? returns true
+ * (null-key), return NIL instead */
+namespace kdb {
+
+  %typemap(out) Key {
+    if ($1.isNull()) {
+      $result = Qnil;
+    } else {
+      $result = SWIG_NewPointerObj(new kdb::Key($1), 
+                                    SWIGTYPE_p_kdb__Key, 
+                                    SWIG_POINTER_OWN | 0);
+    }
+  }
+}
+
+
+/* rename some methods to meet the Ruby naming conventions */
+%rename("name") kdb::Key::getName;
+%rename("name=") kdb::Key::setName;
+
+%rename("base_name") kdb::Key::getBaseName;
+%rename("base_name=") kdb::Key::setBaseName;
+
+%rename("full_name") kdb::Key::getFullName;
+
+%rename("namespace") kdb::Key::getNamespace;
+
+
+/* autorename and templates has some problems */
+%rename("get") kdb::Key::get<std::string>;
+%rename("set") kdb::Key::set<std::string>;
+%alias kdb::Key::get<std::string> "value"
+%alias kdb::Key::set<std::string> "value="
+
+%rename("set_meta") kdb::Key::setMeta<std::string>;
+%rename("get_meta") kdb::Key::getMeta<std::string>;
+
+%alias kdb::Key::setMeta<std::string> "[]="
+%alias kdb::Key::getMeta<std::string> "[]"
+
+/* expose Keys meta KeySet
+ * this allows Ruby-style meta iterator, e.g. k.meta.each ... */
+%feature("autodoc", "allows access to the meta data keyset of the 
+underlaying key, which allows a Ruby-style iteration over metadata:
+  k.meta.each { |m| puts 'meta data: %s: %s' % [m.name, m.value] }
+") kdb::Key::meta;
+%extend kdb::Key {
+  kdb::KeySet* meta() {
+    return new kdb::KeySet($self->getKey()->meta);
+  }
+}
+
+
+/* getMeta Typemap
+ * This is used to convert the input argument to a Ruby string. In certain
+ * cases this is useful, to allow passing in Symbols as meta names. */
+%typemap(in) (const std::string & metaName) {
+  // typemap in for getMeta
+  $input = rb_funcall($input, rb_intern("to_s"), 0, NULL);
+  $1 = new std::string(StringValueCStr($input));
+}
+%typemap(freearg) (const std::string & metaName) {
+  // typemap in for getMeta
+  delete $1;
+}
+
+/* Typemap for setBinary
+ * pass raw data pointer of a Ruby String and its length */
+%typemap(in) (const void * newBinary, size_t dataSize) {
+  $1 = (void *) StringValuePtr($input);
+  $2 = RSTRING_LEN($input);
+}
+
+
+/* 'imitate' va_list as Ruby Hash
+ * 
+ * "missuse" the exception feature of SWIG to provide a custom
```

Comment:
That is why I asked: Exceptions in C++ cause no overhead, unless they are actually thrown; then they are expensive. But it might be irrelevant in relation to what ruby does.